core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -232,6 +232,13 @@ private[spark] class DAGScheduler(
     sc.getConf.getInt("spark.stage.maxConsecutiveAttempts",
       DAGScheduler.DEFAULT_MAX_CONSECUTIVE_STAGE_ATTEMPTS)
 
+  /**
+   * Max stage attempts allowed before a stage is aborted.
+   */
+  private[scheduler] val maxStageAttempts: Int = {
+    Math.max(maxConsecutiveStageAttempts, sc.getConf.get(config.STAGE_MAX_ATTEMPTS))
```

Comment:
Modify this suitably to return an `Option`.

Reply:
Thanks, modified the default value instead.